src/main/java/org/openstreetmap/atlas/geography/atlas/raw/slicing/RawAtlasPointAndLineSlicer.java

```diff
@@ -279,150 +417,294 @@ private void processLineSlices(final Line line, final List<Geometry> slices)
+        {
+            final boolean originalClockwise = new Polygon(line.asPolyLine().truncate(0, 1))
+                    .isClockwise();
+            final boolean sliceClockwise = new Polygon(polylineForSlice.truncate(0, 1))
+                    .isClockwise();
+            if (originalClockwise != sliceClockwise)
+            {
+                polylineForSlice = polylineForSlice.reversed();
+            }
+        }
+        return polylineForSlice;
     }
 
     /**
-     * Given a slice, that is outside the working bound, mark all shape points for this slice for
-     * removal from the final atlas. Since we are removing the slice, we don't need its shape
-     * points.
+     * Helper method for slicing geometry operations that looks at each slice {@link Geometry} and
+     * relates it to the original {@link Geometry}. Should the original entity be entirely within
+     * one country, return true as a short-circuit to avoid further unnecessary operations.
+     * Otherwise, tags each {@link Geometry} with an {@link ISOCountryTag} for the country it's in,
+     * or removes if it's not a valid candidate.
      *
-     * @param slice
-     *            The {@link Geometry} slice being filtered
+     * @param identifier
+     *            The identifier for the original {@link Line} being sliced
+     * @param candidates
+     *            The collection of candidate slice {@link Geometry}
+     * @param results
+     *            The current running collection of valid slice {@link Geometry}
+     * @param target
+     *            The JTS {@link Geometry} for the {@link Line} being sliced
+     * @return True if the entity lays entirely in one country, false otherwise.
      */
-    private void removeShapePointsFromFilteredSliced(final Geometry slice)
+    private boolean relateCandidates(final long identifier,
+            final List<org.locationtech.jts.geom.Polygon> candidates, final List<Geometry> results,
+            final Geometry target)
     {
-        final Coordinate[] jtsSliceCoordinates = slice.getCoordinates();
-        for (final Coordinate coordinate : jtsSliceCoordinates)
+        // Check relation of target to all polygons
+        final Iterator<org.locationtech.jts.geom.Polygon> candidateIterator = candidates.iterator();
+        while (candidateIterator.hasNext())
         {
-            final Location location = JTS_LOCATION_CONVERTER.backwardConvert(coordinate);
-            final Iterable<Point> existingRawAtlasPoints = getStartingAtlas().pointsAt(location);
-            existingRawAtlasPoints
-                    .forEach(point -> this.pointsMarkedForRemoval.add(point.getIdentifier()));
+            final org.locationtech.jts.geom.Polygon candidate = candidateIterator.next();
+            final String countryCode = CountryBoundaryMap.getGeometryProperty(candidate,
+                    ISOCountryTag.KEY);
+            if (Strings.isNullOrEmpty(countryCode))
+            {
+                logger.warn(
```

Comment:
How often does this happen? Would it be helpful to log the polygon here or at least the identifier?

Reply:
good idea! I'll do a second pass through this whole section and make sure there's better logging